File: src/main/java/com/morph/engine/core/IWorld.java
package com.morph.engine.core;

import com.morph.engine.entities.Entity;

import java.util.List;
import java.util.stream.Collectors;

/**
 * Created by Fernando on 1/19/2017.
 */
public interface IWorld {
    Game getGame();
    List<Entity> getEntities();

    default Entity getEntityByName(String name) {
        return getEntities().stream().collect(Collectors.toMap(Entity::getName, e -> e)).get(name);
    }

    default Entity getEntityByID(int id) {
        return getEntities().stream().collect(Collectors.toMap(Entity::getId, e -> e)).get(id);
    }

    boolean addEntity(Entity e); // TODO: Should this really be required?
    boolean removeEntity(Entity e);
}

File: src/main/kotlin/com/morph/engine/core/IWorld.kt
package com.morph.engine.core

import com.morph.engine.entities.Entity

/**
 * Created by Fernando on 1/19/2017.
 */
interface IWorld {
    val game: Game
    val entities : List<Entity>

    fun getEntityByName(name: String): Entity? = entities.find { it.name == name }
    fun getEntityByID(id: Int): Entity? = entities.find { it.id == id }

    fun addEntity(e: Entity): Boolean  // TODO: Should this really be required?
    fun removeEntity(e: Entity): Boolean

    fun init()
    fun destroy() {
        entities.forEach { it.destroy() }
    }
}
